### ISSUE TITLE
Different graphbreaks may make that NumPy code is traced or not

### ISSUE THREAD

**Issue by lezcano** [pytorch/pytorch#109604]
### 🐛 Describe the bug

In the code below, `graph_break` generates two graphs, while using a `print` skips the second graph arguing that `[DEBUG] skipping because no torch.* <resume in fn>`

```python
import torch
import numpy as np


@torch.compile(backend="eager")
def fn(x):
    x = x ** 2
    #torch._dynamo.comptime.graph_break()
    print("HI")
    return 2 * x


fn(np.arange(8))
```

I think this may be caused by the source attribution for ndarrays in dynamo.

cc @mruberry @rgommers @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @ysiraichi @Fidget-Spinner 

### Versions

master

**Comment by ysiraichi** [comment]
As far as I can see, it stops tracing because:

- It thinks it's inside another (not annotated with `torch.compile`) function (due to the graph break)
- After the graph break:
    - There are no allowed `torch` functions
    - `x` is  not a tensor (maybe we could add NumPy arrays to this)

https://github.com/pytorch/pytorch/blob/c2f5d4d8f09b535e434313223434009bad282258/torch/_dynamo/convert_frame.py#L158-L159

**Comment by lezcano** [comment]
Yep, it seems like it's missing adding ndarray to that file. I guess that function is not executed when tracing through the first frame then?

**Comment by atalman** [comment]
It will be included in 2.2.0 release
